crates/chia-protocol/src/spend_bundle.rs

```diff
@@ -94,10 +96,33 @@ impl SpendBundle {
 #[pymethods]
 #[allow(clippy::needless_pass_by_value)]
 impl SpendBundle {
-    #[staticmethod]
+    #[classmethod]
     #[pyo3(name = "aggregate")]
-    fn py_aggregate(spend_bundles: Vec<SpendBundle>) -> SpendBundle {
-        SpendBundle::aggregate(&spend_bundles)
+    fn py_aggregate(cls: &Bound<'_, PyType>, spend_bundles: Vec<Self>) -> PyResult<PyObject> {
+        let aggregated = Self::aggregate(&spend_bundles);
+        Python::with_gil(|py| {
+            // Convert result into potential child class
+            let instance = cls.call(
+                (aggregated.coin_spends, aggregated.aggregated_signature),
+                None,
+            )?;
+
+            Ok(instance.into_py(py))
+        })
+    }
+
+    #[classmethod]
+    #[pyo3(name = "from_parent")]
+    pub fn from_parent(cls: &Bound<'_, PyType>, spend_bundle: Self) -> PyResult<PyObject> {
+        Python::with_gil(|py| {
```

Comment:
I think you can just take a `Python` object as a parameter to get access to the GIL, rather than wrapping everything in `Python::with_gil()`. But maybe you've explored that.

Reply:
Done